src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -1183,23 +1183,42 @@ CanUseExclusiveConnections(Oid relationId, bool localTableEmpty)
 	}
 	else if (shouldRunSequential && ParallelQueryExecutedInTransaction())
 	{
-		char *relationName = get_rel_name(relationId);
-
 		/*
-		 * If there has already been a parallel query executed, the sequential mode
-		 * would still use the already opened parallel connections to the workers,
-		 * thus contradicting our purpose of using sequential mode.
+		 * We decided to use sequential execution. It's either because relation
+		 * has a pre-existing foreign key to a reference table or because we
+		 * decided to use sequential execution due to a query executed in the
+		 * current xact beforehand.
+		 * We have specific error messages for either cases.
 		 */
-		ereport(ERROR, (errmsg("cannot distribute relation \"%s\" in this "
-							   "transaction because it has a foreign key to "
-							   "a reference table", relationName),
-						errdetail("If a hash distributed table has a foreign key "
-								  "to a reference table, it has to be created "
-								  "in sequential mode before any parallel commands "
-								  "have been executed in the same transaction"),
-						errhint("Try re-running the transaction with "
-								"\"SET LOCAL citus.multi_shard_modify_mode TO "
-								"\'sequential\';\"")));
+
+		char *relationName = get_rel_name(relationId);
+
+		if (hasForeignKeyToReferenceTable)
+		{
+			/*
+			 * If there has already been a parallel query executed, the sequential mode
+			 * would still use the already opened parallel connections to the workers,
+			 * thus contradicting our purpose of using sequential mode.
+			 */
+			ereport(ERROR, (errmsg("cannot distribute relation \"%s\" in this "
+								   "transaction because it has a foreign key to "
+								   "a reference table", relationName),
+							errdetail("If a hash distributed table has a foreign key "
+									  "to a reference table, it has to be created "
+									  "in sequential mode before any parallel commands "
+									  "have been executed in the same transaction"),
+							errhint("Try re-running the transaction with "
+									"\"SET LOCAL citus.multi_shard_modify_mode TO "
+									"\'sequential\';\"")));
+		}
+		else if (MultiShardConnectionType == SEQUENTIAL_CONNECTION)
+		{
+			/* FIXME: TODO: I think we should actually return false here or improve error message(/hint) ? */
+			ereport(ERROR, (errmsg("cannot distribute \"%s\" in sequential mode "
+								   "a parallel query was executed in this transaction",
+								   relationName),
+							errhint("TODO: a proper hint ??")));
```

Comment:
Maybe use the hint above:
```
	errhint("If you have manually set "
								"citus.multi_shard_modify_mode to 'sequential', "
								"try with 'parallel' option. If that is not the "
								"case, try distributing local tables when they "
								"are empty."))
```

But, make sure that this hint works to overcome the situation when triggered  (as far  as I  can see yes,  but requires  verification)

Reply:
At this point, we know that table is empty, maybe below ?
```
	errhint("If you have manually set "
		    "citus.multi_shard_modify_mode to 'sequential', "
		    "try with 'parallel' option. "))
```

Reply:
https://github.com/citusdata/citus/pull/4139/commits/0074fd4e09858dfc12ea471399ac5e1f957e0cc1